### ISSUE TITLE
Admin Interface Behavior: Ensuring Consistent Workflow

### ISSUE THREAD

**Issue by maackey** [ZeroK-RTS/Zero-K-Infrastructure#2750]
Is there ever a case when an account should not be deleted/redacted, but inaccessible due to lack of password / Steam link?

Should the admin tools prevent this scenario? Or simply allow with a warning message?

IMO admin tools in particular should be clear in their behavior to prevent accidental issues with user accounts, and this has the potential to inadvertently lock someone out of their account.

**Comment by Licho1** [comment]
I don't undestand what you mean.

**Comment by DeinFreund** [comment]
Admin tools should probably try to warn/prevent that from happening. There are cases where we want to make accounts inaccessible, but they are very rare and can be done via DB.

**Comment by maackey** [comment]
> I don't undestand what you mean.

Hypothetical user is not linked with steam.
Admin comes along and resets password with blank value.
User cannot login with blank password, and cannot log in with steam.

Relatively minor because admin can just set password again to valid temp password, but this can easily be preventable in code by not allowing blank password without steam link.

**Comment by GoogleFrog** [comment]
This doesn't seem like a big deal. A blank password can't be logged onto, but neither can a password that nobody knows.

**Comment by maackey** [comment]
Yeah, its not a big deal (thus the low/minor tags) - but I'd like to remove as much friction as possible from the Admin tools -- whether that is reducing confusion on what buttons/inputs do, making it easier to perform an action -- while keeping potential accidents to a minimum. This would technically remove some functionality, and since it's related to a sensitive area just wanted to be extra careful before changing it. 

I've renamed this issue to expand the scope a bit to the rest of the admin actions, not just change password. I'll go ahead and prevent setting blank password, but I'd also like to remove the now somewhat unnecessary manual username input check -- or at the very least make double checks consistent among the actions with warnings. Delete User Rating seems to have a much bigger impact than Delete Forum Posts -- but doesn't have the same double check. 

Ultimately, considering how the modal already acts as a sanity check, I'm not sure the username field is needed. Any thoughts?

![image](https://user-images.githubusercontent.com/3486898/98101799-b5292b80-1e60-11eb-9066-2016b620185c.png)

![image](https://user-images.githubusercontent.com/3486898/98101850-c6723800-1e60-11eb-90a8-37886b21dcc8.png)
